### PR TITLE
GH-35206: [C++] Look for Conda OpenSSL in Windows verification

### LINKED ISSUE #35206: [CI][Flight][C++] Flight build fails on verification job for windows with missing symbol on GRPC/SSL
### Describe the bug, including details regarding any error messages, version, and platform.

The windows verification job has been failing for the last 15 days:
* [verify-rc-source-windows](https://github.com/ursacomputing/crossbow/actions/runs/4721371081/jobs/8374565556)

It seems that the issue is with GRPC / SSL:

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(1091,5): warning MSB8027: Two or more files with the name of serialization_internal.cc will produce outputs to the same location. This can lead to an incorrect build result.  The files involved are D:\a\crossbow\crossbow\arrow\cpp\src\arrow\flight\serialization_internal.cc, D:\a\crossbow\crossbow\arrow\cpp\src\arrow\flight\transport\grpc\serialization_internal.cc. [D:\a\crossbow\crossbow\arrow\cpp\build\src\arrow\flight\arrow_flight_shared.vcxproj]
  unity_2_cxx.cxx
  unity_1_cxx.cxx
  unity_0_cxx.cxx
  Flight.pb.cc
     Creating library D:/a/crossbow/crossbow/arrow/cpp/build/release/Release/arrow_flight.lib and object D:/a/crossbow/crossbow/arrow/cpp/build/release/Release/arrow_flight.exp
grpc.lib(json_token.cc.obj) : error LNK2019: unresolved external symbol EVP_DigestSignUpdate referenced in function "char * __cdecl compute_and_encode_signature(struct grpc_auth_json_key const *,char const *,char const *)" (?compute_and_encode_signature@@YAPEADPEBUgrpc_auth_json_key@@PEBD1@Z) [D:\a\crossbow\crossbow\arrow\cpp\build\src\arrow\flight\arrow_flight_shared.vcxproj]
grpc.lib(ssl_transport_security.cc.obj) : error LNK2019: unresolved external symbol SSL_get1_peer_certificate referenced in function "enum tsi_result __cdecl ssl_handshaker_result_extract_peer(struct tsi_handshaker_result const *,struct tsi_peer *)" (?ssl_handshaker_result_extract_peer@@YA?AW4tsi_result@@PEBUtsi_handshaker_result@@PEAUtsi_peer@@@Z) [D:\a\crossbow\crossbow\arrow\cpp\build\src\arrow\flight\arrow_flight_shared.vcxproj]
D:\a\crossbow\crossbow\arrow\cpp\build\release\Release\arrow_flight.dll : fatal error LNK1120: 2 unresolved externals [D:\a\crossbow\crossbow\arrow\cpp\build\src\arrow\flight\arrow_flight_shared.vcxproj]
```
This should be fixed for 12.0.0.

### Component(s)

C++, Continuous Integration, FlightRPC

### PR DESCRIPTION
**Pull request by lidavidm**
### Are there any user-facing changes?

* Closes: #35206

### CODE REVIEW

**Comment by lidavidm** [comment]
@github-actions crossbow submit verify-rc-source-windows

File: dev/release/verify-release-candidate.bat [hunk 1]
```diff
@@ -112,6 +112,8 @@ cmake -G "%GENERATOR%" ^
       -DARROW_WITH_ZSTD=ON ^
       -DCMAKE_BUILD_TYPE=%CONFIGURATION% ^
       -DCMAKE_INSTALL_PREFIX=%ARROW_HOME% ^
+      -DCMAKE_PREFIX_PATH=%CONDA_PREFIX%\Lib ^
```
Comment (kou) [inline dev/release/verify-release-candidate.bat:115]: I think that we don't need to append `\Lib` for `CMAKE_PREFIX_PATH`:

```suggestion
      -DCMAKE_PREFIX_PATH=%CONDA_PREFIX% ^
```

https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html

> Each command will add appropriate subdirectories (like `bin`, `lib`, or `include`) as specified in its own documentation.

**Comment by lidavidm** [comment]
@github-actions crossbow submit verify-rc-source-windows

**Comment by lidavidm** [comment]
First attempt didn't work

```
-- Found OpenSSL: optimized;C:/Program Files/OpenSSL/lib/VC/libcrypto64MD.lib;debug;C:/Program Files/OpenSSL/lib/VC/libcrypto64MDd.lib (found suitable version "1.1.1t", minimum required is "1.0.2")  
```

**Comment by lidavidm** [comment]
@github-actions crossbow submit verify-rc-source-windows

File: dev/release/verify-release-candidate.bat [hunk 2]
```diff
@@ -112,6 +112,8 @@ cmake -G "%GENERATOR%" ^
       -DARROW_WITH_ZSTD=ON ^
       -DCMAKE_BUILD_TYPE=%CONFIGURATION% ^
       -DCMAKE_INSTALL_PREFIX=%ARROW_HOME% ^
+      -DCMAKE_PREFIX_PATH=%CONDA_PREFIX% ^
```
Comment (kou) [inline dev/release/verify-release-candidate.bat:115]: It seems that conda's OpenSSL is installed into `%CONDA_PREFIX%\Library\lib`, `%CONDA_PREFIX%\Library\include` and so on:

```suggestion
      -DCMAKE_PREFIX_PATH=%CONDA_PREFIX%\Library ^
```

**Comment by lidavidm** [comment]
@github-actions crossbow submit verify-rc-source-windows

**Comment by lidavidm** [comment]
Thanks Kou, that seems to have been the key thing. I'll rebase out the debug commit once I get off this スーパーおき train

**Comment by lidavidm** [comment]
Thanks Kou, that seems to have been the key thing. I'll rebase out the debug commit once I get off this スーパーおき train

**Comment by kou** [comment]
No problem!
Could you remove `-DOPENSSL_ROOT_DIR=%CONDA_PREFIX%` too? I think that it's needless.

Have a good trip!

**Review by raulcd — APPROVED** [review]
LGTM

**Comment by lidavidm** [comment]
Good find! A little worrying that that much is enough to make CMake use a completely different OpenSSL, but, oh well.

**Pull request merged**